### PR TITLE
Java: Cleanup operation samplers if absent from strategy response

### PR DESCRIPTION
**Pull request by guo0693**
Signed-off-by: jung <jung@uber.com>

## Which problem is this PR solving?
- Fixes [#1728](https://github.com/jaegertracing/jaeger/issues/1728)

## Short description of the changes
- If the server did not return a strategy for operation X, the operation sampler will be removed and the default sampler will be used for operation X

### CODE REVIEW

File: jaeger-core/src/main/java/io/jaegertracing/internal/samplers/PerOperationSampler.java [hunk 1]
```diff
@@ -17,14 +17,13 @@
 import io.jaegertracing.internal.samplers.http.OperationSamplingParameters;
 import io.jaegertracing.internal.samplers.http.PerOperationSamplingParameters;
 import io.jaegertracing.spi.Sampler;
-import java.util.HashMap;
-import lombok.AccessLevel;
-import lombok.AllArgsConstructor;
-import lombok.EqualsAndHashCode;
-import lombok.Getter;
-import lombok.ToString;
+import lombok.*;
```
Comment (vprithvi) [inline jaeger-core/src/main/java/io/jaegertracing/internal/samplers/PerOperationSampler.java]: Seems like an unnecessary change

Comment (guo0693) [reply jaeger-core/src/main/java/io/jaegertracing/internal/samplers/PerOperationSampler.java]: IDE automatically optimized the import.

File: jaeger-core/src/main/java/io/jaegertracing/internal/samplers/PerOperationSampler.java [hunk 2]
```diff
@@ -64,6 +63,20 @@ public synchronized boolean update(OperationSamplingParameters strategies) {
...
+    for (PerOperationSamplingParameters strategy : strategies.getPerOperationStrategies()) {
+      newOps.add(strategy.getOperation());
+    }
+
+    for (String op : operationNameToSampler.keySet()) {
+      if (!newOps.contains(op)) {
+        operationNameToSampler.remove(op);
+        isUpdated = true;
+      }
+    }
```
Comment (vprithvi) [inline jaeger-core/src/main/java/io/jaegertracing/internal/samplers/PerOperationSampler.java]: Could it be simpler to create a new `HashMap<String, GuaranteedThroughputSampler>`, and either insert a sampler update, or create a new sampler; and replace `operationNameToSampler` with this new Map?

File: jaeger-core/src/main/java/io/jaegertracing/internal/samplers/PerOperationSampler.java (hunk 1 above)
Comment (vprithvi) [reply jaeger-core/src/main/java/io/jaegertracing/internal/samplers/PerOperationSampler.java]: It's still an unnecessary change

File: jaeger-core/src/main/java/io/jaegertracing/internal/samplers/PerOperationSampler.java (hunk 2 above)
Comment (guo0693) [reply jaeger-core/src/main/java/io/jaegertracing/internal/samplers/PerOperationSampler.java]: You are right, I thought about recreate the remote sampler using the response and then thought we need to preserve the sampler's internal state, haven't thought about re-insert the sampler to new map. Updating.

File: jaeger-core/src/test/java/io/jaegertracing/internal/samplers/PerOperationSamplerTest.java [hunk 3]
```diff
@@ -169,10 +169,29 @@ public void testUpdateAddOperation() {
...
-                 operationToSamplers.get(OPERATION));
+            undertest.getOperationNameToSampler().get(OPERATION));
   }
 
+  @Test
+  public void testAbsentOperationIsRemoved() {
+    String absentOp = "ShouldBeRemoved";
+    operationToSamplers.put(absentOp, mock(GuaranteedThroughputSampler.class));
+    PerOperationSamplingParameters perOperationSamplingParameters1 =
+            new PerOperationSamplingParameters(OPERATION, new ProbabilisticSamplingStrategy(SAMPLING_RATE));
```
Comment (vprithvi) [inline jaeger-core/src/test/java/io/jaegertracing/internal/samplers/PerOperationSamplerTest.java]: Shouldn't `operationToSamplers` be passed in here to control the starting state of the test?

Comment (guo0693) [reply jaeger-core/src/test/java/io/jaegertracing/internal/samplers/PerOperationSamplerTest.java]: undertest is inited with the operationToSamplers during setup, adding a sampler to the map should do the job regarding the starting state.
